src/v/kafka/protocol/schemata/generator.py

```diff
@@ -1428,13 +1428,40 @@ class response;
 {%- endif %}
 {%- else %}
 {%- if op_type == "request" %}
+{%- if first_flex > 0 %}
+void {{ struct.name }}::encode(response_writer& writer, api_version version) {
+    if (version >= api_version({{ first_flex }})) {
+        writer.write_tags(std::move(unknown_tags));
+    }
+}
+void {{ struct.name }}::decode(request_reader& reader, api_version version) {
```

Comment:
i would have expected to see these changes in `{en,de}code_flex` rather than just `encode`/`decode`. i'm probably just missing some context

Reply:
This is wrapped around some edge case conditional where there are no fields for a struct